reference/opt/shaders-msl/flatten/push-constant.flatten.vert

```diff
@@ -10,18 +10,18 @@ struct PushMe
     float Arr[4];
 };
 
-struct main0_in
-{
-    float4 Pos [[attribute(1)]];
-    float2 Rot [[attribute(0)]];
-};
-
 struct main0_out
 {
     float2 vRot [[user(locn0)]];
     float4 gl_Position [[position]];
 };
 
+struct main0_in
+{
+    float2 Rot [[attribute(0)]];
```

Comment:
The ordering here changed, is that intentional?

Reply:
Yes. It's deliberate.

Previously input structs were ordered by _reverse_ location order...and output structs were ordered by location order. This brings them both inline.

The original requirement was because early versions of Metal _seemed_ to like having that reverse order on the inputs. But I can't find any real requirement for that anymore. It's either been fixed in Metal...or it never existed.

So I have made both in and out structs sort by location so they visually match between vertex and fragment shaders.

Reply:
Ok, the new output makes more sense to me.